### ISSUE TITLE
Improve the Infinity Dimensions UI handling

### ISSUE THREAD

**Issue by kajfik** [IvarK/AntimatterDimensionsSourceCode#957]
From Discord:

- be able to eternity without ID8
- the "Unlock a new dimension" button would be visible only if: player doesn't have ID8 unlocked AND IP < infinity AND "AutoUnlockID" milestone is not reached AND Infinity is broken
- the "Eternity" button would be visible if : IP >= infinity OR "AutoUnlockID" milestone is reached
- An ID row would be visible if: the respective ID is unlocked OR respective ID can be unlocked OR respective ID amount > 0 OR eternities > 0
- An ID row would be semi-transparent if: it's visible AND the respective ID is not unlocked AND the requirement for the respective ID is not reached
- And ID buy button text would be:
  - if unlocked : "Cost: X IP"
  - else if canUnlock() : "Unlock" (or something similar)
  - else : "Reach X AM"

Basically - Eternity button overrides unlock button, and if you really want the unlock at that point, you go to ID tab for it.

Related issues: #722 , #755 , https://github.com/antimatter-dimensions/mobile-issues/issues/25, https://github.com/antimatter-dimensions/mobile-issues/issues/72, https://github.com/antimatter-dimensions/mobile-issues/issues/99
